plugins/linux-capture/pipewire.c

```diff
@@ -410,10 +411,19 @@ static void on_process_cb(void *user_data)
 		}
 
 		g_clear_pointer(&obs_pw->texture, gs_texture_destroy);
-		obs_pw->texture = gs_texture_create_from_dmabuf(
-			obs_pw->format.info.raw.size.width,
-			obs_pw->format.info.raw.size.height, drm_format,
-			GS_BGRX, planes, fds, strides, offsets, modifiers);
+		if (obs_pw->format.info.raw.modifier ==
+		    DRM_FORMAT_MOD_INVALID) {
+			obs_pw->texture = gs_texture_create_from_dmabuf(
+				obs_pw->format.info.raw.size.width,
+				obs_pw->format.info.raw.size.height, drm_format,
+				GS_BGRX, planes, fds, strides, offsets, NULL);
+		} else {
+			obs_pw->texture = gs_texture_create_from_dmabuf(
+				obs_pw->format.info.raw.size.width,
+				obs_pw->format.info.raw.size.height, drm_format,
+				GS_BGRX, planes, fds, strides, offsets,
+				modifiers);
```

Comment:
Can you please use a boolean variable for the invalid check, and continue passing `modifiers` here? I'm thinking of something like this:

```
bool valid_modifiers;

...

valid_modifiers = obs_pw->format.info.raw.modifier != DRM_FORMAT_MOD_INVALID;
obs_pw->texture = gs_texture_create_from_dmabuf(
	obs_pw->format.info.raw.size.width,
	obs_pw->format.info.raw.size.height, drm_format,
	GS_BGRX, planes, fds, strides, offsets,
	valid_modifiers ? modifiers : NULL);
```

Reply:
modifierless is probably the more accurate term, but a added a comment, since there are some synonyms around modifiers.
It's actually nicer with the ternary operator.